sotopia/ui/README.md

```diff
@@ -0,0 +1,141 @@
+# Sotopia UI
+
+## FastAPI Server
+
+The API server is a FastAPI application that is used to connect the Sotopia UI to the Sotopia backend.
+This could also help with other projects that need to connect to the Sotopia backend through HTTP requests.
+
+Here are some initial design of the API server:
+
+### Getting Data from the API Server
+
+#### GET /get/scenarios/{scenario_id}
+
+Get scenarios by scenario_id.
+parameters:
+- scenario_id: str
+
+returns:
+- scenarios: EnvironmentProfile
+
+
+#### GET /get/scenarios
+
+Get all scenarios.
+
+returns:
+- scenarios: list[EnvironmentProfile]
+
+#### GET /get/scenarios/{sceanrio_tag}
+
+Get scenarios by scenario_tag.
+parameters:
+- scenario_tag: str
+(This scenario tag could be a keyword; so people can search for scenarios by keywords)
+
+returns:
+- scenarios: list[EnvironmentProfile]
+
+#### GET /get/agents
+
+Get all agents.
+
+returns:
+- agents: list[AgentProfile]
+
+#### GET /get/agents/{agent_id}
+
+Get agent by agent_id.
+parameters:
+- agent_id: str
+
+returns:
+- agent: AgentProfile
+
+#### GET /get/agents/{agent_gender}
+
+Get agents by agent_gender.
+parameters:
+- agent_gender: Literal["male", "female"]
+
+returns:
+- agents: list[AgentProfile]
+
+#### GET /get/agents/{agent_occupation}
+
+Get agents by agent_occupation.
+parameters:
+- agent_occupation: str
+
+returns:
+- agents: list[AgentProfile]
+
+
+#### GET /get/episodes
+
+Get all episodes.
+
+returns:
+- episodes: list[Episode]
+
+#### GET /get/episodes/{episode_tag}
+
+Get episode by episode_tag.
+parameters:
+- episode_tag: str
+
+returns:
+- episode: list[Episode]
+
+#### GET /get/episodes/{episode_id}
+
+Get episode by episode_id.
+parameters:
+- episode_id: str
+
+returns:
+- episode: Episode
+
+
+### Sending Data to the API Server
+
+#### POST /post/agents/
```

Comment:
I think maybe first starts with `/post/agent`?

Reply:
got it. this makes sense.

Can we do the following three post endpoints?

- POST /agents (json=AgentProfile) create a new agent and return the AgentProfile with id
- PUT /agents/{agent_id} (json=AgentProfile) update an existing agent with the AgentProfile. return the new AgentProfile with id. raise 404 when the agent_id doesn't exist
- GET /agents/{agent_id} similar to the one above, except that it will retrieve the agent instead of updating it.

Reply:
We have get already above

Reply:
Yeah so does the above make sense

Reply:
? what do you mean